### ISSUE TITLE
add Markdown to the phet code style for easier auto formatting

### ISSUE THREAD

**Issue by zepumph** [phetsims/phet-info#216]
From today's dev meeting, this seems helpful and important. I'll export some markdown stuff and add it to our code style

**Comment by zepumph** [comment]
I'll keep working on this. Most importantly we need to fix indented autoformatted bullets like `\n*\n` and others.

**Comment by zepumph** [comment]
https://github.com/phetsims/phet-info/commit/5a74a60a075eec31dca602b0cde7781348f192cf made the auto formatting of md SO MUCH BETTER. I'm really happy with the work above, and I'm ready to come back to devs and tell them to reload their code style. Thanks everyone!

**Comment by zepumph** [comment]
https://github.com/phetsims/sherpa/blob/7624dd4f361370b7b19bd65ed6b92558fe46674b/third-party-licenses.md#L14 is an auto generated file and will just go back to what is was upon next generation (overnight).

**Comment by zepumph** [comment]
Ok @jessegreenberg and @samreid helped me see that https://github.com/phetsims/phet-info/commit/6764df929697784a3208f035a45d8c9e15745e20 was needed to get markdown codestyle all sorted. I'm pretty sure we are now ready for other devs to import it. @samreid will you take another test drive to make sure all is looking right?

**Comment by samreid** [comment]
I imported the new style, formatted center-and-variability/examples.md and still saw these 7 differences.  Not sure if there is trouble or if I'm doing something wrong. We should probably sort this out on a short zoom call:

<details>

````diff

Subject: [PATCH] Update documentation, see https://github.com/phetsims/color-vision/issues/147
---
Index: repos/center-and-variability/examples.md
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/repos/center-and-variability/examples.md b/repos/center-and-variability/examples.md
--- a/repos/center-and-variability/examples.md	(revision 6beb0d55ef481c5737350f4c29edde7513cfb8e7)
+++ b/repos/center-and-variability/examples.md	(date 1694781532932)
@@ -117,15 +117,15 @@
 #### Get the array of soccer ball locations for the indicated screen
 
 ```js
-await phetioClient.invokeAsync('centerAndVariability.medianScreen.model','getDataPoints',[]);
+await phetioClient.invokeAsync('centerAndVariability.medianScreen.model', 'getDataPoints', []);
 ```
 
 ```js
-await phetioClient.invokeAsync('centerAndVariability.meanAndMedianScreen.model','getDataPoints',[]);
+await phetioClient.invokeAsync('centerAndVariability.meanAndMedianScreen.model', 'getDataPoints', []);
 ```
 
 ```js
-await phetioClient.invokeAsync('centerAndVariability.variabilityScreen.model','getDataPoints',[]);
+await phetioClient.invokeAsync('centerAndVariability.variabilityScreen.model', 'getDataPoints', []);
 ```
 
 * Note: on the Variability screen, the array of soccer balls will represent the active player only.
@@ -133,7 +133,7 @@
 #### Get the array of card values, in the order they appear on screen
 
 ```js
-await phetioClient.invokeAsync('centerAndVariability.medianScreen.model.distanceAccordionBox.cardContainerModel','getData')
+await phetioClient.invokeAsync('centerAndVariability.medianScreen.model.distanceAccordionBox.cardContainerModel', 'getData')
 ```
 
 ## Collect current scene data
@@ -178,8 +178,8 @@
 Variability screen. Data for the Median and Mean & Median screens should be obtained using the `phetioID`s listed above.
 
 ```js
-let mean = await phetioClient.invokeAsync( 'centerAndVariability.variabilityScreen.model.selectedSceneModelProperty', 'getValue' )
-        .then(async sceneElement => phetioClient.invokeAsync( sceneElement.phetioID + '.meanValueProperty', 'getValue' ) );
+let mean = await phetioClient.invokeAsync('centerAndVariability.variabilityScreen.model.selectedSceneModelProperty', 'getValue')
+    .then(async sceneElement => phetioClient.invokeAsync(sceneElement.phetioID + '.meanValueProperty', 'getValue'));
 ```
 
 ## Enable or disable soccer balls individually or as a group
@@ -227,7 +227,7 @@
 cards have been manually sorted. For example, the command below will print “hello” upon a successful manual sort:
 
 ```js
-phetioClient.invoke('centerAndVariability.medianScreen.model.distanceAccordionBox.cardContainerModel.manuallySortedEmitter','addListener',[()=>console.log('hello')])
+phetioClient.invoke('centerAndVariability.medianScreen.model.distanceAccordionBox.cardContainerModel.manuallySortedEmitter', 'addListener', [() => console.log('hello')])
 ```
 
 Note: The emitter fires every time the cards have been sorted manually. The Sort Data checkbox auto-sorts the cards, and

````

</details>

**Comment by zepumph** [comment]
@samreid can you please reapply the code style again. I believe things are more correct now. I also just committed another minor change since likely designers that are editing CAV examples aren't auto formatting (which is totally fine).

**Comment by samreid** [comment]
I reapplied the code style, restarted IntelliJ, reformatted center-and-variability examples.md, and saw this result:

<details>

```diff

Subject: [PATCH] Rename migration processors, see https://github.com/phetsims/phet-io-wrappers/issues/572
---
Index: repos/center-and-variability/examples.md
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/repos/center-and-variability/examples.md b/repos/center-and-variability/examples.md
--- a/repos/center-and-variability/examples.md	(revision 1d232b147a3e0387af3e0631de6c5f49c6b2e01d)
+++ b/repos/center-and-variability/examples.md	(date 1697845829037)
@@ -117,15 +117,15 @@
 #### Get the array of soccer ball locations for the indicated screen
 
 ```js
-await phetioClient.invokeAsync('centerAndVariability.medianScreen.model','getDataPoints',[]);
+await phetioClient.invokeAsync( 'centerAndVariability.medianScreen.model', 'getDataPoints', [] );
 ```
 
 ```js
-await phetioClient.invokeAsync('centerAndVariability.meanAndMedianScreen.model','getDataPoints',[]);
+await phetioClient.invokeAsync( 'centerAndVariability.meanAndMedianScreen.model', 'getDataPoints', [] );
 ```
 
 ```js
-await phetioClient.invokeAsync('centerAndVariability.variabilityScreen.model','getDataPoints',[]);
+await phetioClient.invokeAsync( 'centerAndVariability.variabilityScreen.model', 'getDataPoints', [] );
 ```
 
 * Note: on the Variability screen, the array of soccer balls will represent the active player only.
@@ -133,7 +133,7 @@
 #### Get the array of card values, in the order they appear on screen
 
 ```js
-await phetioClient.invokeAsync('centerAndVariability.medianScreen.model.distanceAccordionBox.cardContainerModel','getData')
+await phetioClient.invokeAsync( 'centerAndVariability.medianScreen.model.distanceAccordionBox.cardContainerModel', 'getData' )
 ```
 
 ## Collect current scene data
@@ -179,7 +179,7 @@
 
 ```js
 let mean = await phetioClient.invokeAsync( 'centerAndVariability.variabilityScreen.model.selectedSceneModelProperty', 'getValue' )
-        .then(async sceneElement => phetioClient.invokeAsync( sceneElement.phetioID + '.meanValueProperty', 'getValue' ) );
+  .then( async sceneElement => phetioClient.invokeAsync( sceneElement.phetioID + '.meanValueProperty', 'getValue' ) );
 ```
 
 ## Enable or disable soccer balls individually or as a group
@@ -227,7 +227,7 @@
 cards have been manually sorted. For example, the command below will print “hello” upon a successful manual sort:
 
 ```js
-phetioClient.invoke('centerAndVariability.medianScreen.model.distanceAccordionBox.cardContainerModel.manuallySortedEmitter','addListener',[()=>console.log('hello')])
+phetioClient.invoke( 'centerAndVariability.medianScreen.model.distanceAccordionBox.cardContainerModel.manuallySortedEmitter', 'addListener', [ () => console.log( 'hello' ) ] )
 ```
 
 Note: The emitter fires every time the cards have been sorted manually. The Sort Data checkbox auto-sorts the cards, and

```

**Comment by zepumph** [comment]
Hmmm, I am not sure what is happening. I just re-exported my settings, pasted them onto the phet-code-style, and there were no changes, I then auto formatted CAV examples and didn't see what you are seeing. Perhaps try on your side one more time, and then from there we should pair on this.

**Comment by samreid** [comment]
I applied the style again (to a new style configuration), then pressed format in CAV examples.md and saw 

![image](https://github.com/phetsims/phet-info/assets/679486/e5c2e752-394b-4c0b-8fea-5b98eee07716)

![image](https://github.com/phetsims/phet-info/assets/679486/0fc4c6af-7597-4925-a71b-dbe431906d01)

Isn't this formatting result correct since it is adding the whitespace after open parentheses? Let's chat.

**Comment by zepumph** [comment]
@samreid can you please paste a picture this settings page to compare. I just noticed that I don't get text highlighting for snippets I've marked as "```js":

I wonder if I need a plugin or something to be smarter:

![image](https://github.com/phetsims/phet-info/assets/6856943/c60cd75f-5a43-4972-b183-9c055658179e)

**Comment by samreid** [comment]
Here is my copy:

![image](https://github.com/phetsims/phet-info/assets/679486/a1dc3ec1-68d7-47d8-97a5-59c0ff802cd7)

**Comment by zepumph** [comment]
We did end up cracking the case, it looks like for my webstorm instance, the setting "inject languages in code fences" doesn't work, even though it is checked. We confirmed that this does work on @samreid's intellij and @matthew-blackman's webstorm. 

TODO:
* I will try to fix that bug on my side by reinstalling the markdown plugin.
* @samreid will run the auto formatter on all `.md` files in the project and commit them here so that code fences are formatted.
* We will tell all devs to reload the code style and start using auto formatting on md files. 
* We will close this issue.
